src/workers/event.rs

```diff
@@ -700,69 +646,188 @@ where
-                self.handle_actor_stopped(&message.lattice_id, actor).await
-            }
-            Event::HostHeartbeat(host) => {
-                self.handle_host_heartbeat(&message.lattice_id, host).await
-            }
-            Event::HostStarted(host) => self.handle_host_started(&message.lattice_id, host).await,
-            Event::HostStopped(host) => self.handle_host_stopped(&message.lattice_id, host).await,
-            Event::LinkdefDeleted(_ld) => {
-                // TODO: This will need to be handled when we start managing applications
-                Ok(())
-            }
-            Event::ProviderStarted(provider) => {
-                self.handle_provider_started(&message.lattice_id, provider)
-                    .await
-            }
-            Event::ProviderStopped(provider) => {
-                self.handle_provider_stopped(&message.lattice_id, provider)
-                    .await
-            }
-            Event::ProviderHealthCheckPassed(ProviderHealthCheckPassed { data, host_id }) => {
-                self.handle_provider_health_check(&message.lattice_id, host_id, data, false)
-                    .await
-            }
-            Event::ProviderHealthCheckFailed(ProviderHealthCheckFailed { data, host_id }) => {
-                self.handle_provider_health_check(&message.lattice_id, host_id, data, true)
-                    .await
-            }
-            Event::ManifestPublished(_data) => {
-                // TODO: Will need to be handled for managing applications
-                Ok(())
-            }
-            Event::ManifestUnpublished(_data_manifest) => {
-                // TODO: Will need to be handled for managing applications
-                Ok(())
+            Event::ActorStarted(actor) => self
+                .handle_actor_started(&message.lattice_id, actor)
+                .await
+                .map(|_| {
+                    actor
+                        .annotations
+                        .get(APP_SPEC_ANNOTATION)
+                        .map(|s| s.as_str())
+                }),
+            Event::ActorStopped(actor) => self
+                .handle_actor_stopped(&message.lattice_id, actor)
+                .await
+                .map(|_| {
+                    actor
+                        .annotations
+                        .get(APP_SPEC_ANNOTATION)
+                        .map(|s| s.as_str())
+                }),
+            Event::HostHeartbeat(host) => self
+                .handle_host_heartbeat(&message.lattice_id, host)
+                .await
+                .map(|_| None),
+            Event::HostStarted(host) => self
+                .handle_host_started(&message.lattice_id, host)
+                .await
+                .map(|_| None),
+            Event::HostStopped(host) => self
+                .handle_host_stopped(&message.lattice_id, host)
+                .await
+                .map(|_| None),
+            Event::LinkdefDeleted(_ld) => Ok(None),
```

Comment:
We will need to care about this with the linkdef scaler, but I can add it in #84